integration_test.go

```diff
@@ -161,3 +161,72 @@ func TestAllowedRegexpProxy(t *testing.T) {
 	// TODO(oskar) - gracefully stop proxy by the end of the test
 	es.Stop(ctx)
 }
+
+func TestBasicAuthRoundtrip(t *testing.T) {
+	tc := newTestClient()
+
+	echoServerAddr := "localhost:3010"
+	es := newEchoServer(echoServerAddr)
+
+	ctx := context.Background()
+
+	payload := fmt.Sprintf("{%q:%q}", "hello", "world")
+	payloadBytes := []byte(payload)
+
+	go es.Start()
+
+	// Authentication Proxy setup
+	username := "username"
+	password := "password"
+
+	authProxyConfig := ProxyConfig{
+		RemoteAddress:              "http://" + echoServerAddr,
+		ShouldValidateRequests:     true,
+		AuthenticationScheme:       BasicAuthScheme,
+		AllowedBasicAuthUserString: fmt.Sprintf("%s:%s", username, password),
+	}
+	authProxyAddr := "localhost:3012"
+
+	proxy, err := NewProxy(&authProxyConfig)
+
+	require.NoError(t, err)
+
+	go func() {
+		t.Log(http.ListenAndServe(authProxyAddr, proxy))
```

Comment:
Will this go routine ever exit?

Reply:
It did at the end of the tests, however now that is done explicitly by stopping the server with:
```
	authProxyServer.Shutdown(ctx)
	signingProxyServer.Shutdown(ctx)
```